src/Connection.ts

```diff
@@ -95,6 +96,22 @@ interface ServerInfo {
 	/** Multivalue database server limits */
 	limits: DbServerLimits;
 }
+
+interface RequestOptions {
+	requestId?: string | null;
```

Comment:
I don't think we should allow `null` here.  When doing the destructure with defaulting, `null` will not trigger falling back to the default value (i.e. the random UUID) and would instead pass `null` through.  I don't think we want `null` as the header value ever, so the easier approach to resolve that would be to disallow `null` as a valid type here, just leaving it as optional.